### PR TITLE
[macOS] platform_channel sample style cleanups

### PR DESCRIPTION
**Pull request by cbracken**
Applies a few Swift style cleanups to the macOS platform_channel sample.

Also adds ChannelName, BatteryState, ErrorCode enums to bring the example a bit closer to the iOS Swift example. Will follow up with a patch that applies relevant style fixes to the iOS Swift example.

No test changes since app semantics have not changed.

Issue: https://github.com/flutter/flutter/issues/79204

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/platform_channel/macos/Runner/MainFlutterWindow.swift [hunk 1]
```diff
@@ -5,37 +5,54 @@
...
+enum BatteryState {
+  static let charging = "charging"
+  static let discharging = "discharging"
+  static let unavailable = "UNAVAILABLE"
+}
+
+enum ErrorCode {
+  static let noBattery = "NO_BATTERY"
+  static let unavailable = "UNAVAILABLE"
+}
```
Comment (hellohuanlin) [inline examples/platform_channel/macos/Runner/MainFlutterWindow.swift:22]: these can be string backed enums. For example:

```
enum BatteryState: String {
  case charging
  case discharging
  case unavailable
}

enum ErrorCode: String {
  case noBarry = "NO_BATTERY"
  case unavailable = "UNAVAILABLE" 
}
```

**Review by hellohuanlin — APPROVED** [review]
May need to request test exempt to land. The unit test is not needed since it's example code.

File: examples/platform_channel/macos/Runner/PowerSource.swift [hunk 2]
```diff
@@ -12,25 +12,25 @@ enum PowerState {
 }
 
 /// A convenience wrapper for an IOKit power source.
-class PowerSource {
-  let info = IOPSCopyPowerSourcesInfo().takeRetainedValue()
-  lazy var sources = IOPSCopyPowerSourcesList(info).takeRetainedValue() as Array
+final class PowerSource {
+  private let info = IOPSCopyPowerSourcesInfo().takeRetainedValue()
+  private lazy var sources = IOPSCopyPowerSourcesList(info).takeRetainedValue() as Array
```
[the remark this answers is not in the document]
Comment (cbracken) [reply examples/platform_channel/macos/Runner/PowerSource.swift:17]: It won't change; but yeah I'd need to declare an explicit initialiser due to the use of self. Will do that.

File: examples/platform_channel/macos/Runner/MainFlutterWindow.swift (hunk 1 above)
Comment (cbracken) [reply examples/platform_channel/macos/Runner/MainFlutterWindow.swift]: Added to make it similar to the iOS swift example, but it's inconsistent with the rest of the samples, irrelevant to demonstrating channels, and makes it so the useful code is all hidden below the fold, and you've got to scroll down to get to the actual bit that we want to demonstrate. Deleted instead for similarities to all the other samples, bar iOS Swift.

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Pull request merged**